catroid/src/main/java/org/catrobat/catroid/content/eventids/NfcEventId.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Catroid: An on-device visual programming system for Android devices
+ * Copyright (C) 2010-2018 The Catrobat Team
+ * (<http://developer.catrobat.org/credits>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * An additional term exception under section 7 of the GNU Affero
+ * General Public License, version 3, is available at
+ * http://developer.catrobat.org/license_additional_term
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.catrobat.catroid.content.eventids;
+
+import com.google.common.base.Objects;
+
+public class NfcEventId extends EventId {
+	final String tag;
+
+	public NfcEventId(String tag) {
+		this.tag = tag;
+	}
+
+	@Override
+	public boolean equals(Object o) {
+		if (this == o) {
+			return true;
+		}
+		if (o == null || getClass() != o.getClass()) {
```

Comment:
is there a reason you use ```getClass() != o.getClass() ``` instead of ``` !o instanceof NfcEventId``` ? I'm not saying you should change it, I would just like to hear your reasons 😉

Reply:
Actually I just let Android Studio autogenerate it.
I would probably also use instanceof, because it provides nullcheck and class check in one. The only difference I would expect is the behaviour when this class gets inherited.

Reply:
sure, makes sense. so do you want it to behave this way when it is inherited? -> anyway, only food for thought anyway. all fine for now as far as I'm concerned. I'll merge it.

Reply:
once you rebased it of course 😉

Reply:
Now that I think about it, it makes sense that two objects can only be equal if they have the exact same class. 
Ofc it depends on what you want to do, but intuitively this is exactly what I would expect equals to do.